### ISSUE TITLE
[4.9.4] SF_SWITCH_FALLTHROUGH false positive

### ISSUE THREAD

**Issue by ben-manes** [spotbugs/spotbugs#3617]
Upgrading from 4.9.3 and the rule misinterprets a `continue` as a fall-through scenario.

<img width="1087" height="393" alt="Image" src="https://github.com/user-attachments/assets/d29bb868-7c8d-45a2-a6ad-c5b82e5cd39f" />

The method [scheduleAfterWrite()](https://github.com/ben-manes/caffeine/blob/38dd616b945749edcc102642adb1b08274ae3a88/caffeine/src/main/java/com/github/benmanes/caffeine/cache/BoundedLocalCache.java#L1591-L1620) has a `continue` on the `PROCESSING_TO_IDLE` case,

```java
/**
 * Conditionally schedules the asynchronous maintenance task after a write operation. If the
 * task status was IDLE or REQUIRED then the maintenance task is scheduled immediately. If it
 * is already processing then it is set to transition to REQUIRED upon completion so that a new
 * execution is triggered by the next operation.
 */
void scheduleAfterWrite() {
  @Var int drainStatus = drainStatusOpaque();
  for (;;) {
    switch (drainStatus) {
      case IDLE:
        casDrainStatus(IDLE, REQUIRED);
        scheduleDrainBuffers();
        return;
      case REQUIRED:
        scheduleDrainBuffers();
        return;
      case PROCESSING_TO_IDLE:
        if (casDrainStatus(PROCESSING_TO_IDLE, PROCESSING_TO_REQUIRED)) {
          return;
        }
        drainStatus = drainStatusAcquire();
        continue;
      case PROCESSING_TO_REQUIRED:
        return;
      default:
        throw new IllegalStateException("Invalid drain status: " + drainStatus);
    }
  }
}
```

**Comment by ben-manes** [comment]
The rule fails if the `continue` is replaced by a `break`, which has the same effect in this logic but I thought was also a little less clear on the intent. The rule passes if using a `return` which is invalid.

**Comment by xzel23** [comment]
It also fails for new switch syntax when a case is a no-op.

Example code taken from my project, the shown case is reported as fall-through:
```
                XMLEvent event = reader.nextEvent();

                switch (event) {
                    ...
                    case XMLEvent xmlEvent when xmlEvent.getEventType()  == XMLStreamConstants.SPACE -> {
                        // do nothing - skip whitespace
                    }
                    ...
                }
```

**Comment by gtoison** [comment]
This should be fixed by #3624 
@xzel23 so far I wasn't able to reproduce your false positive, so probably a different problem

**Comment by xzel23** [comment]
Note that the switch uses a guarded pattern (i.e., a 'when' clause). I think this might be an important detail, as the generated bytecode probably will look to spotbugs like this:

```
    case XMLEvent xmlEvent:
        if (xmlEvent.getEventType()  == XMLStreamConstants.SPACE) {
            break;
        }
        // fallthrough
```

**Comment by gtoison** [comment]
@xzel23 I've moved your issue to #3627 
The problem here is with a switch inside a loop (and the `continue` within the switch).
This issue should be closed once #3624 is merged.

I have noted that you're using the `when` clause but so far I wasn't able to reproduce the false positive.
It would be great if you could provide a sample reproducing the problem.
